api/v1beta1/controlplane_types.go

```diff
@@ -56,6 +56,8 @@ type ControlPlaneList struct {
 	Items           []ControlPlane `json:"items"`
 }
 
+// +kubebuilder:validation:XValidation:message="ControlPlane requires an image to be set on controller container",rule="has(self.deployment.podTemplateSpec) && has(self.deployment.podTemplateSpec.spec.containers) && self.deployment.podTemplateSpec.spec.containers.exists(c, c.name == 'controller' && has(c.image))"
```

Comment:
I think we should have some testing strategy for those CEL rules. It's easy to make even simple mistakes in a plain text DSL included in a comment. Is it possible to cover them with K8s envtest? It should be quite lightweight in terms of resources and execution times, but robust

Reply:
I propose to use kcfg `crdsvalidation` package for this purpose. https://github.com/Kong/kubernetes-configuration/pull/220 exports this (up until now the types there have been located under `crdsvalidation_test` package). When that gets merged I'll work on using it for changes here.

Reply:
Added https://github.com/Kong/gateway-operator/pull/984/commits/20d8936afbc595b0c7c10f4a49008aae6247d151. 

Uses commit from https://github.com/Kong/kubernetes-configuration/pull/221.